corehq/apps/hqwebapp/static/hqwebapp/js/inactivity.js

```diff
@@ -164,11 +184,19 @@ log("session successfully extended, hiiding warning popup");
         }, 100, {trailing: false}));
         $("body").on("keypress mousemove", _.debounce(function () {
             keyboardOrMouseActive = false;
-            if (warningActive) {
+            if (shouldShowWarning) {
                 showWarningModal();
             }
         }, 500));
 
+        // Send no-op request to server to extend session when there's client-side user activity on this page.
+        // _.throttle will prevent this from happening too often.
+        var keepAliveTimeout = 60 * 1000;
+log("page loaded, will send a keep-alive request to server every click/keypress, at most once every " + (keepAliveTimeout / 1000 / 60) + " minutes");
+        $("body").on("keypress click", _.throttle(function () {
+            extendSession();
+        }, keepAliveTimeout));
```

Comment:
How did you settle on using keypress and click events as indicators of activity?  Know if that's standard practice on sites that log you out frequently?  I don't see anything wrong with it, just wondering, since it seems if anything overgenerous.

Reply:
I don't know if there's a standard, this is my imagination. The tricky piece is coming up with something that's reasonable for both HQ and web apps, when they have different architectures. The session is already extended on most ajax requests, which works reasonably well on HQ but doesn't work for long forms in formplayer, where the user only interacts with formplayer, not HQ. So the other approach I considered was hooking into formplayer requests, making those extend the session. Frankly, it was simpler to key off of clicks and typing, and it's easier for users to understand.